src/serializer.ts

```diff
@@ -3,10 +3,18 @@ import { Transaction } from "../generated/transaction_pb";
 import { FiatAmount } from "../generated/fiat_amount_pb";
 import { XRPAmount } from "../generated/xrp_amount_pb";
 import { Currency } from "../generated/currency_pb";
+import Utils from "./utils";
 
 /* Allow `any` since this class doing progressive conversion of protocol buffers to JSON. */
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
+interface PaymentJSON {
+  Amount: object | string;
```

Comment:
Support for Fiat payments is kinda around: 
https://github.com/xpring-eng/xpring-common-protocol-buffers/blob/master/proto/payment.proto

I think there are some optimizations that I might try to make to the internal protocol buffer structure here. 

Because of the current protocol buffer format we need to support xrp and fiat here. I'm not particularly interested in doing fiat stuff at this time - it's low priority given the other things I'm working on. 

I think there's some long term thinking about whether or not these two amounts really need to be different in the client side API  (i.e. using `string` for XRP is just shorthand for a fiat amount).  That's probably worth a longer discussion, possibly with a document and with consensus from the rippled team. 

Shorter term (this week) my goal is to just drop protocol buffers from the public APIs completely so that we can make internal refactors and changes as needed.

Reply:
I don't think "Fiat" is the right term here. We call them "[issued currencies](https://xrpl.org/issued-currencies-overview.html)", "issuances", or "IOUs". These issued currencies can represent anything of value, including digital currencies like Bitcoin or Ether, other assets like precious metals, or other forms of value.

For more details see:
- [Consensus Principles and Rules](https://xrpl.org/consensus-principles-and-rules.html)
- [Data Types - Currencies](https://xrpl.org/basic-data-types.html#currencies)
- [Become an XRP Ledger Gateway](https://xrpl.org/become-an-xrp-ledger-gateway.html)

Reply:
(By the way, one of the "improvements" that ripple-lib made on the rippled API was to always [use an object](https://xrpl.org/rippleapi-reference.html#currency) for an amount; unlike the rippled API which uses a string for amounts of XRP drops. I don't care either way -- in fact I kind of prefer rippled's way -- but I thought I'd mention this just in case.)

Reply:
Noted - I filed an issue for this in Asana. I've been thinking about restructuring this for a bit. 

The initial bit is going to be refactoring these objects out of the public API, which I'm currently working on.